query-engine/js-connectors/smoke-test-js/src/connector/neon.ts

```diff
@@ -49,16 +49,18 @@ type NeonConfig = PoolConfig;
 
 class PrismaNeon implements Connector, Closeable {
   private pool: Pool
-  private maybeVersion?: string
+  private versionPromise: Promise<string>
   private isRunning: boolean = true
   flavor = "postgres"
 
   constructor(config: NeonConfig) {
     this.pool = new Pool(config)
     // lazily retrieve the version and store it into `maybeVersion`
-    this.pool.query('SELECT VERSION()').then((results) => {
-      this.maybeVersion = results.rows[0]['version']
-    })
+    this.versionPromise = new Promise((resolve, reject) => {
+      this.pool.query('SELECT VERSION()')
+        .then((results) => resolve(results.rows[0]['version']))
+        .catch((error) => reject(error));
```

Comment:
This `catch` will run for errors that happen in the `resolve` callback and not just in `this.pool.query`, should this be the second argument to `then` instead?

Or better yet, could we replace the whole statement with

```ts
this.versionPromise = this.pool.query('SELECT VERSION()')
  .then((results) => resolve(results.rows[0]['version']))`
```

or is there a reason why we are wrapping it in another promise?

Reply:
No, I think we could define it the way you suggest, if `resolve` is a defined symbol within the closure.

Reply:
that was a copy-paste artifact, sorry, I meant
```ts
this.versionPromise = this.pool.query('SELECT VERSION()')
  .then((results) => results.rows[0]['version'])`
```